src/muxer.ts

```diff
@@ -512,42 +549,58 @@ export class Muxer<T extends Target> {
 			}
 
 			track.currentChunk = {
-				startTimestamp: sample.timestamp,
+				startTimestamp: sample.pts,
 				samples: []
 			};
 		}
 
 		track.currentChunk.samples.push(sample);
 	}
 
-	#validateTimestamp(timestamp: number, track: Track) {
+	#validateTimestamp(pts: number, dts: number, track: Track) {
 		// Check first timestamp behavior
-		if (this.#options.firstTimestampBehavior === 'strict' && track.lastTimestamp === -1 && timestamp !== 0) {
+		if (this.#options.firstTimestampBehavior === 'strict' && track.lastDTS === -1 && dts !== 0) {
 			throw new Error(
-				`The first chunk for your media track must have a timestamp of 0 (received ${timestamp}). Non-zero ` +
+				`The first chunk for your media track must have a timestamp of 0 (received DTS=${dts}). Non-zero ` +
 				`first timestamps are often caused by directly piping frames or audio data from a MediaStreamTrack ` +
 				`into the encoder. Their timestamps are typically relative to the age of the document, which is ` +
 				`probably what you want.\n\nIf you want to offset all timestamps of a track such that the first one ` +
 				`is zero, set firstTimestampBehavior: 'offset' in the options.\n`
 			);
-		} else if (this.#options.firstTimestampBehavior === 'offset') {
-			if (track.firstTimestamp === undefined) {
-				track.firstTimestamp = timestamp;
+		} else if (
+			this.#options.firstTimestampBehavior === 'offset' ||
+			this.#options.firstTimestampBehavior === 'cross-track-offset'
+		) {
+			if (track.firstDTS === undefined) {
+				track.firstDTS = dts;
+			}
+
+			let baseDTS = track.firstDTS;
+			if (this.#options.firstTimestampBehavior === 'cross-track-offset') {
+				// Since each track may have its firstDTS set independently, but the tracks' timestamps come from the
+				// same clock, we should subtract the earlier of the (up to) two tracks' first timestamps to ensure A/V
+				// sync.
+				baseDTS = Math.min(
+					...[this.#audioTrack, this.#videoTrack]
+						.filter(x => (x?.firstDTS ?? null) !== null)
+						.map(x => x.firstDTS)
+				);
 			}
```

Comment:
It'd write it like this:
```ts
let baseDTS: number;

if (this.#options.firstTimestampBehavior === 'offset') {
    baseDTS = track.firstDTS;
} else {
    // Since each track may have its firstDTS set independently, but the tracks' timestamps come from the
    // same clock, we should subtract the earlier of the (up to) two tracks' first timestamps to ensure A/V
    // sync.
    baseDTS = Math.min(
        this.videoTrack?.firstDTS ?? Infinity,
        this.audioTrack?.firstDTS ?? Infinity
    );
}
```

Reads a bit more clearly and it's a bit more efficient.